src/daemon/daemonaccesscontrol.h

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef DAEMONSESSION_H
+#define DAEMONSESSION_H
+
+#include <QString>
+
+class QLocalSocket;
+
+class DaemonAccessControl final {
+ public:
+  explicit DaemonAccessControl();
+  ~DaemonAccessControl();
+
+  bool isSessionActive() const;
+
+  bool startSession(QLocalSocket* socket = nullptr);
+  void resetSession();
+
+  bool authorizeCommandForPeer(const QString& command,
```

Comment:
Since this returns a bool, is something like isCommandAuthorizedForPeer a better name?